controls/V-71945.rb

```diff
@@ -1,79 +1,71 @@
-fail_interval=900
+auth required pam_faillock.so preauth silent audit deny=3 even_deny_root unlock_time=604800 fail_interval=900
+auth [default=die] pam_faillock.so authfail audit deny=3 even_deny_root unlock_time=604800 fail_interval=900
+account required pam_faillock.so
+
+If the \"even_deny_root\" setting is not defined on both lines with the
+\"pam_faillock.so\" module name, this is a finding.
+
+# grep pam_faillock.so /etc/pam.d/system-auth-ac
+auth required pam_faillock.so preauth silent audit deny=3 even_deny_root unlock_time=604800 fail_interval=900
+auth [default=die] pam_faillock.so authfail audit deny=3 even_deny_root unlock_time=604800 fail_interval=900
+account required pam_faillock.so
 
 If the \"even_deny_root\" setting is not defined on both lines with the
 \"pam_faillock.so\" module name, this is a finding."
-  tag "fix": "Configure the operating system to automatically lock the root account
-until the locked account is released by an administrator when three unsuccessful
-logon attempts in 15 minutes are made.
+  tag "fix": "Configure the operating system to automatically lock the root
+account until the locked account is released by an administrator when three
+unsuccessful logon attempts in 15 minutes are made.
 
 Modify the first three lines of the auth section of the
-\"/etc/pam.d/system-auth-ac\" and \"/etc/pam.d/password-auth-ac\" files to match the
-following lines:
+\"/etc/pam.d/system-auth-ac\" and \"/etc/pam.d/password-auth-ac\" files to
+match the following lines:
 
-auth        required       pam_faillock.so preauth silent audit deny=3
-even_deny_root fail_interval=900 unlock_time=604800
+auth        required       pam_faillock.so preauth silent audit deny=3 even_deny_root fail_interval=900 unlock_time=604800
 auth        sufficient     pam_unix.so try_first_pass
-auth        [default=die]  pam_faillock.so authfail audit deny=3 even_deny_root
-fail_interval=900 unlock_time=604800
+auth        [default=die]  pam_faillock.so authfail audit deny=3 even_deny_root fail_interval=900 unlock_time=604800
+account required pam_faillock.so
+
+Note: Any updates made to \"/etc/pam.d/system-auth-ac\" and
+\"/etc/pam.d/password-auth-ac\" may be overwritten by the \"authconfig\"
+program. The \"authconfig\" program should not be used."
+  tag "fix_id": "F-78297r2_fix"
 
-and run the \"authconfig\" command."
+  required_lines = [
+    'auth required pam_faillock.so even_deny_root',
+    'auth sufficient pam_unix.so try_first_pass',
+    'auth [default=die] pam_faillock.so even_deny_root'
+  ]
 
   describe pam('/etc/pam.d/password-auth') do
-    required_lines = [
-      'auth required pam_faillock.so even_deny_root',
-      'auth sufficient pam_unix.so try_first_pass',
-      'auth [default=die] pam_faillock.so even_deny_root'
-    ]
+    its('lines') { should match_pam_rules(required_lines) }
+    its('lines') { should match_pam_rule('auth .* pam_faillock.so (preauth|authfail)').all_with_args('even_deny_root') }
```

Comment:
You can modify the `required_lines` above to include the regex so that you don't need multiple tests.

Reply:
@trevor-vaughan Part of the reason I included this pattern is to say that "if you have any other line using this module (using preauth or authfail), it must also use `even_deny_root`"  Do you think that's excessive?

Reply:
@samcornwell It doesn't really hurt anything but there may be reasons that someone has added rules without that (I can't think of any but PAM can be fun).

Reply:
@trevor-vaughan So I can leave it? If so this is good to go, I've gone ahead and removed the constant attributes.

Reply:
@samcornwell Yeah, we can leave it.